include/nccl_ofi_ep_addr_list.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_EP_ADDR_LIST_H
+#define NCCL_OFI_EP_ADDR_LIST_H
+
+struct nccl_ofi_ep_addr_list;
+typedef struct nccl_ofi_ep_addr_list nccl_ofi_ep_addr_list_t;
+
+/* Endpoint structure used by plugin code */
+struct nccl_net_ofi_ep;
+typedef struct nccl_net_ofi_ep nccl_net_ofi_ep_t;
+
+/**
+ * Initialize an endpoint-address-set pair list. This function allocates memory
+ * for the new list
+ *
+ * nccl_ofi_ep_addr_list_fini should be called to free memory associated with
+ * this list.
+ *
+ * @param list: output, pointer to the new list
+ */
+void nccl_ofi_init_ep_addr_list(nccl_ofi_ep_addr_list_t **list);
```

Comment:
I didn't notice this before (sorry), but these should really be named `<prefix>_<object>_<action>`.  So this should really be `nccl_ofi_ep_addr_list_init()`.  And likewise for all the other functions.

Reply:
Also, why have this as the function signature?  Either it should return an error code or return a pointer to the allocated thing, but this option just seems hard to use for no good reason.

Reply:
Agreed, and agreed.